ionoscloud/resource_server_vcpu.go

```diff
@@ -0,0 +1,355 @@
+package ionoscloud
+
+import (
+	"context"
+	"fmt"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/validation"
+	"github.com/ionos-cloud/terraform-provider-ionoscloud/v6/utils"
+)
+
+func resourceVCPUServer() *schema.Resource {
+	return &schema.Resource{
+		CreateContext: resourceVCPUServerCreate,
+		ReadContext:   resourceServerRead,
+		UpdateContext: resourceServerUpdate,
+		DeleteContext: resourceServerDelete,
+		Importer: &schema.ResourceImporter{
+			StateContext: resourceServerImport,
+		},
+		CustomizeDiff: checkServerImmutableFields,
+
+		Schema: map[string]*schema.Schema{
+			"name": {
+				Type:             schema.TypeString,
+				Required:         true,
+				ValidateDiagFunc: validation.ToDiagFunc(validation.StringIsNotWhiteSpace),
+			},
+			"cores": {
+				Type:     schema.TypeInt,
+				Required: true,
+			},
+			"ram": {
+				Type:     schema.TypeInt,
+				Required: true,
+			},
+			"availability_zone": {
+				Type:             schema.TypeString,
+				Optional:         true,
+				Computed:         true,
+				ValidateDiagFunc: validation.ToDiagFunc(validation.StringInSlice([]string{"AUTO", "ZONE_1", "ZONE_2"}, true)),
+			},
+			"boot_volume": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"boot_cdrom": {
+				Type:     schema.TypeString,
+				Optional: true,
+			},
+			"cpu_family": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"type": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"boot_image": {
+				Type:     schema.TypeString,
+				Computed: true,
+				Optional: true,
+			},
+			"primary_nic": {
+				Type:        schema.TypeString,
+				Computed:    true,
+				Description: "Id of the primary network interface",
+			},
+			"primary_ip": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"firewallrule_id": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"firewallrule_ids": {
+				Type:       schema.TypeList,
+				Optional:   true,
+				Computed:   true,
+				ConfigMode: schema.SchemaConfigModeAttr,
+				Elem: &schema.Schema{
+					Type: schema.TypeString,
+				},
+			},
+			"datacenter_id": {
+				Type:             schema.TypeString,
+				Required:         true,
+				ForceNew:         true,
+				ValidateDiagFunc: validation.ToDiagFunc(validation.StringIsNotWhiteSpace),
```

Comment:
can be isuuid instead

Reply:
@cristiGuranIonos done in 35a13f8951cffce1c4c25e74f6242197e6cf8354